cysync/src/store/provider/transactionStatusProvider/txnStatusExecutor.ts

```diff
@@ -0,0 +1,259 @@
+  near as nearServer,
+  serverBatch as batchServer,
+  solana as solanaServer
+} from '@cypherock/server-wrapper';
+import { flatMap } from 'lodash';
+
+import logger from '../../../utils/logger';
+import { transactionDb } from '../../database';
+
+import { TxnStatusItem } from './txnStatusItem';
+
+export interface RequestMetaProcessInfo {
+  meta: IRequestMetadata[];
+  item: TxnStatusItem;
+  error?: Error;
+  isFailed: boolean;
+}
+
+export interface ExecutionResult {
+  item: TxnStatusItem;
+  isFailed: boolean;
+  isComplete?: boolean;
+  canRetry?: boolean;
+  error?: Error;
+  processResult?: any;
+  delay?: number;
+}
+
+export const getRequestsMetadata = (
+  item: TxnStatusItem
+): IRequestMetadata[] => {
+  const coin = COINS[item.coinType];
+
+  if (!coin) {
+    logger.warn('Invalid coin in sync queue', {
+      coinType: item.coinType
+    });
+    return undefined;
+  }
+
+  if (coin instanceof BtcCoinData) {
+    const metadata = bitcoinServer.transaction
+      .transactionStatus({ coinType: coin.abbr, hash: item.txnHash })
+      .getMetadata();
+    return [metadata];
+  }
+
+  if (coin instanceof EthCoinData) {
+    const metadata = ethServer.transaction
+      .transactionStatus(
+        { network: coin.network, txHash: item.txnHash },
+        item.isRefresh
+      )
+      .getMetadata();
+    return [metadata];
+  }
+
+  if (coin instanceof NearCoinData) {
+    const metadata = nearServer.transaction
+      .transactionStatus(
+        {
+          network: coin.network,
+          txHash: item.txnHash,
+          sender: item.sender
+        },
+        item.isRefresh
+      )
+      .getMetadata();
+    return [metadata];
+  }
+
+  if (coin instanceof SolanaCoinData) {
+    const metadata = solanaServer.transaction
+      .transactionStatus(
+        { network: coin.network, txHash: item.txnHash },
+        item.isRefresh
+      )
+      .getMetadata();
+    return [metadata];
+  }
+
+  logger.warn('Invalid coin in sync queue', {
+    coinType: item.coinType
+  });
+  return [];
+};
+
+// process the response of one request item based on the coinType
+export const processResponses = async (
+  item: TxnStatusItem,
+  response: batchServer.IBatchResponse
+): Promise<any> => {
+  const coin = COINS[item.coinType];
+
+  if (!coin) {
+    throw new Error('Invalid coin in balance sync item: ' + item.coinType);
+  }
+
+  let status;
+
+  if (coin instanceof EthCoinData) {
+    if (response.data.result?.status === '1') status = Status.SUCCESS;
+    else if (response.data.result?.status === '0') status = Status.FAILURE;
+    else status = Status.PENDING; // Pending or not found
+  } else if (coin instanceof NearCoinData) {
+    // if invalid transaction hash, result: not present and error
+    // if success, result.status.successValue; if failure, result.status.Failure; if unprocessed, result.status.Unknown
+    if (response.data.error !== undefined)
+      throw new Error(`Error from api ${response.data.error}`);
```

Comment:
If this is an error, why are we not returning error response in the server?

Reply:
is it good to process on server side?

Reply:
move this check to server so the error is in the response header itself.

Reply:
done